src/sentry/utils/snuba.py

```diff
@@ -889,7 +903,24 @@ def _raw_snql_query(
             logger.info(f"{referrer}.body: {query}")
             query = query.set_debug(True)
 
-        body = query.snuba()
+        with thread_hub.start_span(
+            op="snuba_snql_validate", description=f"validate query {referrer}"
+        ):
+            scope = thread_hub.scope
+            if scope.transaction:
+                query.set_parent_api(scope.transaction.name)
+
+            metrics.incr(
+                "snuba.parent_api",
+                tags={
+                    "parent_api": query.parent_api.name
```

Comment:
Is there a specific reason, the `parent_api` tag over here is set differently than how it is set on line 851?

Reply:
Yes, the `query` here is an instance of `snuba_sdk.Query` whereas on line 851 `query_params` is a `Mapping[str, Any]`. One is calling Snuba with SnQL and the other is calling Snuba with the legacy endpoint.